cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -791,6 +837,38 @@ func (p *BicepProvider) purgeKeyVaults(
 	return nil
 }
 
+func (p *BicepProvider) purgeCognitiveAccounts(
+	ctx context.Context,
+	resourceGroup string,
+	cognitiveAccounts []armcognitiveservices.AccountsClientGetResponse,
+	options DestroyOptions,
+	skip bool,
+) error {
+	for _, cogAccount := range cognitiveAccounts {
+		accountName := cogAccount.Name
```

Comment:
nit: I could see us being more tolerant of contractual violations from the service here as the client.

The service gave us an invalid response, and it seems just as  appropriate here to just ignore it versus stopping the entire delete operation.

Reply:
IIRC, the data contact generated by azsdk-for-go comes from swagger API definitions. So the pointer here means that the official swagger API definition lists the fields as optional. Yet, they're not always semantically correct. Most obvious cases are on things like Subscriptions where the ID is listed as optional. See #1577.

Reply:
I feel better just checking the pointers before using them to avoid a panic.  Hopefully we can improve all these cases when addressing #1577